toast-tk-runtime/src/main/java/io/toast/tk/runtime/AbstractProjectRunner.java

```diff
@@ -152,11 +178,111 @@ protected void createAndOpenReport(final ITestPlan project) {
+					failure = testPage.getTestFailureNumber();
+					total = success + failure;
+					String name = testPage.getName();
+					
+					sb.append("Test report for "+ name + "\n").append(System.lineSeparator());
+					sb.append("Total number of steps : " + total + "\n");
+					sb.append("Number of Success steps in test : " + success + "\nNumber of Failure steps in test : " + failure + "\n\n");
+
+				}
+				message.setText(sb.toString());
+				Transport.send(message);
+				System.out.println("Message sent successfully");
+			}catch (MessagingException mex) {
+				mex.printStackTrace();
+			}
+		}
+	}
+
+
+	protected void sendEmailReport (final ITestPlan project) {
+		
+		final Properties prop = new Properties();
+
+		InputStream resourceAsStream = this.getClass().getResourceAsStream("/toast.properties");
+
+		try (final Reader resourceFileReader = new InputStreamReader(resourceAsStream)) {
+			prop.load(resourceFileReader);
+		} catch (final IOException e) {
+			LOG.error(e.getMessage(), e);
+		}
+		
+		for (final ICampaign campaign : project.getCampaigns()) {
+			String mailTo = prop.getProperty("mail.user.to");
+			String mailFrom = prop.getProperty("mail.user.from");
+			String password = prop.getProperty("mail.password");
+			String host = prop.getProperty("mail.host");
+			String port = prop.getProperty("mail.port");
+			String ssl = prop.getProperty("mail.ssl");
+			String tls = prop.getProperty("mail.tls");
+			
+			Properties properties = System.getProperties();
+			properties.setProperty("mail.smtp.host", host);
+			prop.put("mail.smtp.socketFactory.port", port);
+			prop.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+			prop.put("mail.smtp.socketFactory.fallback", "false");
+			prop.setProperty("mail.smtp.starttls.enable", ssl);
+	       prop.setProperty("mail.smtp.password", password );
+	        prop.setProperty("mail.smtp.auth", "true");
+			Session session = Session.getDefaultInstance(properties);
+			int success = 0;
+			int failure = 0;
+			int total = 0;
+			try {
+				MimeMessage message = new MimeMessage(session);
+				message.setFrom(new InternetAddress(mailFrom));
+				message.addRecipient(Message.RecipientType.TO, new InternetAddress(mailTo));
+				message.setSubject("Test report");
+				StringBuilder sb = new StringBuilder();
+				
+				for (final ITestPage testPage : campaign.getTestCases()) {
+					success = testPage.getTestSuccessNumber();
+					failure = testPage.getTestFailureNumber();
+					total = success + failure;
+					String name = testPage.getName();
+					
+					sb.append("Test report for "+ name + "\n").append(System.lineSeparator());
+					sb.append("Total number of steps : " + total + "\n");
+					sb.append("Number of Success steps in test : " + success + "\nNumber of Failure steps in test : " + failure + "\n\n");
+
+				}
+				message.setText(sb.toString());
+				Transport.send(message);
+				System.out.println("Message sent successfully");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AS106)